src/test/java/jenkins/branch/RateLimitBranchPropertyTest.java

```diff
@@ -169,13 +170,14 @@ public void rateLimitsBlockBuilds(int rate) throws Exception {
             );
             assertThat(master.isInQueue(), is(false));
             assertThat(master.getQueueItem(), nullValue());
+            assertThat(master.getBuilds().getLastBuild(), isNotNull());
+            long startTime = master.getBuilds().getLastBuild().getTimeInMillis();
```

Comment:
This is probably why it flakes all the time lastBuild is not null so it uses the time of the last build to throttle
https://github.com/jenkinsci/branch-api-plugin/blob/a1c604b25e6d7aa704f82a0138699cee32388d07/src/main/java/jenkins/branch/RateLimitBranchProperty.java#L507-L530